src/OpenSage.Game/Logic/Object/Locomotor.cs

```diff
@@ -97,11 +107,32 @@ public void MoveTowardsPosition(in TimeInterval gameTime, in Vector3 targetPoint
 
             // When we get to minimum braking distance, start braking.
             var delta = targetPoint - transform.Translation;
+
             // Distance is 2D
             var distanceRemaining = delta.Vector2XY().Length();
+            var braking = GetLocomotorValue(x => x.Braking);
+
+            switch (_locomotorTemplate.Appearance)
+            {
+                case LocomotorAppearance.Treads:
+                    break;
+                default:
+                    if (nextPoint != null) braking = 0; //
+
+                    // Calculate turn radius from speed & turnrate
+                    var circumference = 360.0f / GetTurnRate() * oldSpeed;
```

Comment:
There's no explicit **TurnRadius** in C&C Generals, but the calculation here should be correct